esphome/components/mq/sensor.py

```diff
@@ -0,0 +1,230 @@
+    CONF_SENSOR_PROPANE: MQGasType.MQ_GAS_TYPE_PROPANE,
+    CONF_SENSOR_SMOKE: MQGasType.MQ_GAS_TYPE_SMOKE,
+    CONF_SENSOR_TOLUENO: MQGasType.MQ_GAS_TYPE_TOLUENO,
+}
+
+MQSensor = mq_ns.class_("MQSensor", cg.PollingComponent)
+
+
+def get_model_valid_sensors(config):
+    return set(MQ_MODEL_SENSORS[config[CONF_MODEL]])
+
+
+def get_sensors_schemas(config):
+    return set(
+        list(
+            dict(
+                filter(lambda elem: "sensor_" in elem[0].lower(), config.items())
+            ).keys()
+        )
+    )
+
+
+def validate_sensors(config):
+    model = config[CONF_MODEL]
+    valid_sensors = get_model_valid_sensors(config)
+    sensors_schemas = get_sensors_schemas(config)
+    if sensors_schemas - valid_sensors != set():
+        invalid_sensors = sensors_schemas.difference(valid_sensors)
+        raise cv.Invalid(
+            "Invalid sensors definition for {model}\nNot supported sensors:\n\t{}\n{model} supports:\n\t{}"
+            "".format(
+                ", ".join(map(str, invalid_sensors)),
+                ", ".join(map(str, valid_sensors)),
+                model=model,
+            )
+        )
+    return config
+
+
+sensor_schema = sensor.sensor_schema(
+    unit_of_measurement=UNIT_PARTS_PER_MILLION,
+    icon=ICON_MOLECULE_CO2,
+    accuracy_decimals=2,
+    device_class=DEVICE_CLASS_CARBON_MONOXIDE,
+    state_class=STATE_CLASS_MEASUREMENT,
+)
+
+CONFIG_SCHEMA = cv.All(
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(MQSensor),
+            cv.Required(CONF_PIN): pins.internal_gpio_analog_pin_schema,
+            cv.Optional(CONF_RL, default=10.0): cv.positive_float,
+            cv.Optional(CONF_R0): cv.positive_float,
+            cv.Required(CONF_MODEL): cv.enum(MQ_MODELS, upper=True, space="_"),
+            cv.Optional(CONF_SENSOR_ACETONA): sensor_schema,
+            cv.Optional(CONF_SENSOR_ALCOHOL): sensor_schema,
+            cv.Optional(CONF_SENSOR_BENZENE): sensor_schema,
+            cv.Optional(CONF_SENSOR_CH4): sensor_schema,
+            cv.Optional(CONF_SENSOR_CL2): sensor_schema,
+            cv.Optional(CONF_SENSOR_CO): sensor_schema,
+            cv.Optional(CONF_SENSOR_CO2): sensor_schema,
+            cv.Optional(CONF_SENSOR_ETHANOL): sensor_schema,
+            cv.Optional(CONF_SENSOR_H2): sensor_schema,
+            cv.Optional(CONF_SENSOR_HEXANE): sensor_schema,
+            cv.Optional(CONF_SENSOR_HYDROGEN): sensor_schema,
+            cv.Optional(CONF_SENSOR_ISO_BUTANO): sensor_schema,
+            cv.Optional(CONF_SENSOR_LPG): sensor_schema,
+            cv.Optional(CONF_SENSOR_NH4): sensor_schema,
+            cv.Optional(CONF_SENSOR_NOX): sensor_schema,
+            cv.Optional(CONF_SENSOR_O3): sensor_schema,
+            cv.Optional(CONF_SENSOR_PROPANE): sensor_schema,
+            cv.Optional(CONF_SENSOR_SMOKE): sensor_schema,
+            cv.Optional(CONF_SENSOR_TOLUENO): sensor_schema,
+        }
+    ).extend(cv.polling_component_schema("10s")),
```

Comment:
```suggestion
    ).extend(cv.polling_component_schema("60s")),
```

I'd strongly suggest to use 60s as the default update interval for consistency unless there's a reason you really need lower ones.